File: events.cpp
#include<iostream>

using namespace std;

int main(int argc, char ** argv)
{

    int e1s,e1e,e2s,e2e;
    cout<<"\nEnter starting and ending point for Event 1=";
    cin>>e1s>>e1e;
    cout<<"\nEnter starting and ending point for Event 2=";
    cin>>e2s>>e2e;
    if(e2s>e1e || e1s>e2e )
    {
        cout<<"\nNo Overlap";
    }
    else if(e2s>e1s && e1e>e2e || e1s>e2s && e2e>e1e)
    {
        cout<<"\nFull Overlap";
    }
    else
    {
        cout<<"\nPartial Overlap";
    }
    
}

File: events.java
import java.util.*;

public class events
{
    public static void main(String[] args)
    {
        int e1s,e1e,e2s,e2e;
        Scanner scn = new Scanner(System.in);
        System.out.println("\nEnter starting and ending of event 1=>");
        e1s=scn.nextInt();
        e1e=scn.nextInt();
        System.out.println("\nEnter starting and ending of event 2=>");
        e2s=scn.nextInt();
        e2e=scn.nextInt();

    if(e2s>e1e || e1s>e2e )
    {
        System.out.println("\nNo Overlap");
    }
    else if(e2s>e1s && e1e>e2e || e1s>e2s && e2e>e1e)
    {
        System.out.println("\nFull Overlap");
    }
    else
    {
        System.out.println("\nPartial Overlap");
    }

    }
}